grails-web-url-mappings/src/main/groovy/grails/web/mapping/ResponseRedirector.groovy

```diff
@@ -91,6 +91,13 @@ class ResponseRedirector {
             absolute = (absoluteArgument == null) ? true : (Boolean.TRUE == absoluteArgument)
         }
 
+        // When redirecting from UrlMappings the original request params are on webRequest.originalParams
+        // instead of arguments.params
+        def webRequest = GrailsWebRequest.lookup()
```

Comment:
Could you change this to `lookup(HttpServletRequest request)` instead as it will be more efficient and avoid the thread local lookup

Reply:
Done!